app/src/org/commcare/network/GetAndParseActor.java

```diff
@@ -0,0 +1,141 @@
+package org.commcare.network;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import org.commcare.CommCareApp;
+import org.commcare.CommCareApplication;
+import org.commcare.core.interfaces.HttpResponseProcessor;
+import org.commcare.core.network.AuthInfo;
+import org.commcare.core.network.AuthenticationInterceptor;
+import org.commcare.core.network.ModernHttpRequester;
+import org.commcare.util.LogTypes;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.services.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+
+/**
+ * Created by amstone326 on 5/8/18.
+ */
+
+public abstract class GetAndParseActor {
+
+    protected static final String APP_ID = "app_id";
+    protected static final String DEVICE_ID = "device_id";
+    protected static final String APP_VERSION = "app_version";
+    protected static final String CC_VERSION = "cc_version";
+    protected static final String QUARANTINED_FORMS_PARAM = "num_quarantined_forms";
```

Comment:
Are these last 3 applicable to all GetAndParseActors? Seems like those are only heartbeat related.